### DISCUSSION TITLE
A script that installs Active Backup for Business Agent on any machine

### DISCUSSION THREAD

**Discussion by Pinki0529** [tteck/Proxmox#3000] [Ideas]
a script that installs Active Backup for Business Agent on any machine,
including the required components
and system requirements.
Is it possible to create such a script?

**Comment by Mellowlynx** [comment]
Think Ansible is the way to go for this.

**Comment by Pinki0529** [comment]
I was more interested in a backup of the entire Proxmox host, which can be restored via lan. And it is an incremental copy. Besides, in special cases, you can quickly run it as a VM on your synology server.

**Comment by Mellowlynx** [comment]
I read it as you want to add it to all the VM's and LXC's.
But I get your point that you would like to be able to spin it up on your Synology server, and install in on the nodes.
I still think that an Ansible playbook is the way to go. 
Next to the installation, you will be able to make the needed config changes to point it to the right directory and/or server, where to save the backups. When you get a new node, you run the playbook on there too, and it's added to your backup infrastructure in minutes.

I doubt that tteck would make such a script, but maybe someone (or you) can do that and add it as a pull request.

Myself, I do not back up my hosts.
There are almost no changes to the configs on them, and what I do like to change, .motd, adding my SSH keys, etc. I deploy that using an Ansible playbook I made.

**Comment by Pinki0529** [comment]
Unfortunately, my knowledge of Linux is at a basic level.

I have never successfully completed installing Active Backup Agent on Linux.

Such a script would be helpful not only for the Proxmox user, but also for the entire group of people using Linux and wanting to make a backup copy of their computer and more.

All my computers are backed up to NAS, except the Linux ones.
I would like to, and it would be nice to have such a copy, because in the event of any failure, I can easily restore everything in various ways in a short time.

Therefore, your proposal to learn software that is unknown to me from scratch does not interest me at all. I'm sorry.

Maybe someone with the appropriate knowledge will one day create such a script and it will be helpful not only to me, for which I thank you in advance.
